### DISCUSSION TITLE
Inverse Design

### DISCUSSION THREAD

**Discussion by AbhijithMoni** [mdolab/pyoptsparse#314] [Q&A] [answered]
I would like to know is it possible to do inverse design study with pyoptsparse. For example, inverse design case of an airfoil: initial shape (NACA0012) and  optimize it to final shape (RAE2822) with the target Cp (Coefficient of pressure). 

Thanks

**Comment by marcomangano** [comment, marked as answer]
Hello Abhijith,
pyOptSparse is a Python wrapper for a set of optimization algorithms, so your question goes beyond this code specific domain. If you are able to formulate the inverse design as an optimization problem with your optimizer of choice then you can use pyOptSparse in your script. Check [this page of the documentation](https://mdolab-pyoptsparse.readthedocs-hosted.com/en/latest/guide.html) to learn about the problem formulation used by pyOptSparse and its API.
